src/main/scala/com/advancedtelematic/campaigner/daemon/DeviceInstallationReportListener.scala

```diff
@@ -19,17 +19,18 @@ class DeviceInstallationReportListener()(implicit db: Database, ec: ExecutionCon
   val campaigns = Campaigns()
 
   def apply(msg: DeviceInstallationReport): Future[Unit] = {
-    val resultStatus = if (msg.result.success)
-      DeviceStatus.successful
-    else
-      DeviceStatus.failed
+    val (resultStatus, failureCode) =
+      if (msg.result.success)
+        DeviceStatus.successful -> None
+      else
+        DeviceStatus.failed -> Some(msg.result.code)
```

Comment:
What is in `msg.result.code` when an installation succeeds? Do you think it's worth storing this code in the table not only when an installation failed?

Reply:
I would suggest to always store the code. I believe it was original idea that a code is always present and that is why the `success` flag was introduced - to distinguish between success and failure. I also can imagine different "success" codes

Reply:
As we discussed, I split the `finishDevice` method in three different methods to succeed/fail/cancel, and we save the `resultCode` when `DeviceStatus.success` or `DeviceStatus.failed`.